rust/private/rust_analyzer.bzl

```diff
@@ -214,7 +235,10 @@ def _rust_analyzer_detect_sysroot_impl(ctx):
 
 rust_analyzer_detect_sysroot = rule(
     implementation = _rust_analyzer_detect_sysroot_impl,
-    toolchains = ["@rules_rust//rust:toolchain"],
+    toolchains = [
+        "@rules_rust//rust:toolchain",
+        "@rules_rust//rust/rust_analyzer:toolchain_type",
```

Comment:
why not name this `"@rules_rust//rust/rust_analyzer:toolchain_type"` for consistency with `"@rules_rust//rust:toolchain"`?

Reply:
I think `@rules_rust//rust:toolchain` is inconsistent with other Bazel rules ([rules_cc](https://github.com/bazelbuild/rules_cc/blob/0.0.2/cc/BUILD#L33-L36), [rules_python](https://github.com/bazelbuild/rules_python/blob/0.10.2/python/BUILD)). I care more that the community is consistent more than whatever the value in this PR. What would you recommend happen here?

Reply:
Thanks! Then should we consider renaming `@rules_rust//rust:toolchain` to `@rules_rust//rust:toolchain_type` (as a follow-up)?

Reply:
Created https://github.com/bazelbuild/rules_rust/pull/1479